indico/modules/events/timetable/views/__init__.py

```diff
@@ -129,4 +129,6 @@ def inject_meeting_body(event, **kwargs):
 
 def _entry_title_key(entry):
     obj = entry.object
-    return obj.full_title if entry.type == TimetableEntryType.SESSION_BLOCK else obj.title
+    if entry.type == TimetableEntryType.SESSION_BLOCK:
+        return (obj.code, obj.full_title)
```

Comment:
Mixing tuples and single strings isn't a problem? To me it feels safer to do this:

```
entries.sort(key=lambda entry: (entry.start_dt, *_entry_title_key(entry)))
```

And then always return a tuple, either single-element with code+fulltitle if there's a code, otherwise just fulltitle/title.

Reply:
It seems that nested tuples as sort keys do work, so probably the unpacking is not required